csrc/ops/arith.cpp

```diff
@@ -657,6 +657,31 @@ TensorView* imag(TensorView* tv) {
   return imag(tv->as<Val>())->as<TensorView>();
 }
 
+// construct complex tensor from real and imag tensors
+Val* complex(Val* r, Val* i) {
+  DataType dtype = r->getDataType().value();
+  TORCH_CHECK(
+      dtype == i->getDataType().value(),
+      "real and imag data type should be same in complex().");
+  DataType complex_dtype;
+  BinaryOpType complex_op;
+  if (dtype == DataType::Double) {
+    complex_dtype = DataType::ComplexDouble;
+    complex_op = BinaryOpType::ComplexDouble;
+  } else if (dtype == DataType::Float) {
+    complex_dtype = DataType::ComplexFloat;
+    complex_op = BinaryOpType::ComplexFloat;
+  } else {
+    TORCH_CHECK(false, "complex() only supports float and double types");
+  }
```

Comment:
nit: please add a `getComplexTypeFromType` similar to https://github.com/NVIDIA/Fuser/blob/9f9ba02d2d4aab6587ca76c3869baec4299c9459/csrc/type.h#L208

Reply:
done.